### PR TITLE
add context.GetAs()

### LINKED ISSUE #89: Add context.GetAs(target *interface{}) error
When storing custom structs in the context (to avoid global state) it's common to use similar code

```go
v, err := ctx.Get(appKey{})
if err != nil {
    t.Error(err)
    return ctx
}

app := v.(sapp.App)
app = app.Unhealthy()
```

The syntax would look similar to `json.Unmarshal()`. The above code would look like this:

```go
var app sapp.App
err := ctx.GetAs(appKey{}, &app)
if err != nil {
    t.Error(err)
    return ctx
}

app = app.Unhealthy()
```

### LINKED ISSUE #112: Generate slice getters/setters in the context
**Is your feature request related to a problem? Please describe.**
I'd like to manage slices in the context easily.

**Describe the solution you'd like**
Generate slice variants for the current set of primitive types.

**Describe alternatives you've considered**
Using plain Get is quite cumbersome at the moment. `GetAs` could be an alternative as proposed in #89

### PR DESCRIPTION
**Pull request by bkielbasa**
Fixes #89 
Fixes #112

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-linters-settings:
+xlinters-settings:
```
Comment (sagikazarmark) [inline .golangci.yml:1]: Is this change intentional?

File: context.go [hunk 2]
```diff
@@ -49,6 +50,21 @@ func (ctx Context) Get(key interface{}, defaultValue ...interface{}) (interface{
...
 }
 
+// GetAs copies data from tke key to the dest.
+// Supports maps, slices and structs.
+func (ctx Context) GetAs(key interface{}, dest interface{}) error {
+	if _, ok := ctx.values[key]; !ok {
+		return fmt.Errorf("the key %+v does not exist", key)
+	}
+
+	d, err := json.Marshal(ctx.values[key])
```
Comment (sagikazarmark) [inline context.go:60]: How about using [mapstructure](https://github.com/mitchellh/mapstructure)?

**Review by sagikazarmark — APPROVED** [review]

File: context.go (hunk 2 above)
Comment (bkielbasa) [reply context.go:60]: I assumed that the value can be anything and I want to keep everything as it was originally. It can be a string, int, struct, map, or anything else. So, if you set a struct of a specific type to the context, you should be able to get it back as well. The mapstructure is an additional dependency and I don't see any benefits of using it over the JSON package from the std lib.

File: .golangci.yml (hunk 1 above)
Comment (bkielbasa) [reply .golangci.yml:1]: no, I think that golangci did it somehow...
